frontend/src/queries/nodes/DataTable/DataTableExport.tsx

```diff
@@ -71,23 +204,59 @@ export function DataTableExport({ query }: DataTableExportProps): JSX.Element |
-                                  actor={isPersonsNode(query.source) ? 'persons' : 'events'}
-                                  limit={EXPORT_MAX_LIMIT}
-                              >
-                                  <LemonButton fullWidth status="stealth">
-                                      Export all columns
-                                  </LemonButton>
-                              </ExportWithConfirmation>,
-                          ]
-                        : []
-                ),
+                ]
+                    .concat(
+                        canExportAllColumns
+                            ? [
+                                  <ExportWithConfirmation
+                                      key={0}
+                                      placement={'bottomRight'}
+                                      onConfirm={() => startDownload(query, false)}
+                                      actor={isPersonsNode(query.source) ? 'persons' : 'events'}
+                                      limit={EXPORT_MAX_LIMIT}
+                                  >
+                                      <LemonButton fullWidth status="stealth">
+                                          Export all columns
+                                      </LemonButton>
+                                  </ExportWithConfirmation>,
+                              ]
+                            : []
+                    )
+                    .concat([
+                        <LemonDivider key={2} />,
+                        <LemonButton
+                            key={3}
+                            fullWidth
+                            status="stealth"
+                            onClick={() => {
+                                if (dataTableRows) {
+                                    copyTableToCsv(
+                                        dataTableRows,
+                                        columnsInResponse ?? columnsInQuery,
+                                        queryWithDefaults
+                                    )
+                                }
+                            }}
+                        >
+                            Copy CSV to clipboard
+                        </LemonButton>,
+                        <LemonButton
+                            key={3}
+                            fullWidth
+                            status="stealth"
+                            onClick={() => {
+                                if (dataTableRows) {
+                                    copyTableToJson(
+                                        dataTableRows,
+                                        columnsInResponse ?? columnsInQuery,
+                                        queryWithDefaults
+                                    )
+                                }
+                            }}
+                        >
+                            Copy JSON to clipboard
+                        </LemonButton>,
```

Comment:
Let's be defensive and wrap a "isPersonsNode() || isEventQuery() || ..." condition around these lines. Who knows what other nodes the export button will show up on, and having links that won't do anything will be more frustrating than not having them at all.